crates/virtio-queue/src/queue.rs

```diff
@@ -1500,4 +1506,45 @@ mod tests {
         assert_eq!(q.try_set_size(15).unwrap_err(), Error::InvalidSize);
         assert_eq!(q.size(), expected_val)
     }
+
+    #[test]
+    // This is a regression test for a fuzzing finding. If the driver requests a reset of the
+    // device, but then does not re-initializes the queue then a subsequent call to process
+    // a request should yield no descriptors to process. Before this fix we were processing
+    // descriptors that were added to the queue before, and we were ending up processing 255
+    // descriptors per chain.
+    fn test_regression_timeout_after_reset() {
+        // The input below was generated by libfuzzer and adapted for this test.
+        let m = &GuestMemoryMmap::<()>::from_ranges(&[(GuestAddress(0x0), 0x10000)]).unwrap();
+        let vq = MockSplitQueue::new(m, 1024);
+
+        // This input below was generated by the fuzzer.
+        let descriptors: Vec<Descriptor> = vec![
+            Descriptor::new(21508325467, 0, 1, 4),
+            Descriptor::new(2097152, 4096, 3, 0),
+            Descriptor::new(18374686479672737792, 4294967295, 65535, 29),
+            Descriptor::new(76842670169653248, 1114115, 0, 0),
+            Descriptor::new(16, 983040, 126, 3),
+            Descriptor::new(897648164864, 0, 0, 0),
+            Descriptor::new(111669149722, 0, 0, 0),
+        ];
+        vq.build_multiple_desc_chains(&descriptors).unwrap();
+
+        let mut q: Queue = vq.create_queue().unwrap();
+        q.reset();
+        q.set_ready(true);
```

Comment:
Should we add a test for the reverse, i.e. set_avail_addr is setting the available ring address != 0, but the ready flag is false?

Reply:
I can extend this test to account for that.

Reply:
Extended the test! @lauralt @rbradford please re-review.